### PR TITLE
Use maven enforcer plugin to clean up dependencies

### PR DESCRIPTION
**Pull request by jackjlli**
This PR adds maven enforcer plugin to clean up dependencies.
The command to find out dependenciy conflicts is:
```mvn enforcer:enforce```

### CODE REVIEW

**Comment by kishoreg** [comment]
Amazing work. Thanks for cleaning this up.  will this allow us to get rid of shaded jars for pinot-hadoop and pinot-common? cc @jenniferdai

**Comment by jackjlli** [comment]
@kishoreg If we wanna clean up some jars during the build, maybe `Maven clean plugin` would be a better option? https://maven.apache.org/plugins/maven-clean-plugin/

**Comment by kishoreg** [comment]
@jackjlli I think thats available by default and we do clean install during our build.

**Comment by jackjlli** [comment]
@kishoreg We can finally make the enforce command pass by removing `jersey-client:1.19.2:test:jar` out of the root pom. And `jersey-client:1.19.2:test:jar` is only used in pinot-server module. Now I'm checking whether we can downgrade this test jar version to `1.9` for pinot-server.

**Comment by jackjlli** [comment]
@kishoreg @jenniferdai Now when running `mvn clean install`, enforcer plugin will be triggered for every module:
```
$ mvn clean -DskipTests install | grep "enforcer-plugin"
[INFO] --- maven-enforcer-plugin:3.0.0-M2:enforce (enforce-dependency-convergence) @ pinot-common ---
[INFO] --- maven-enforcer-plugin:3.0.0-M2:enforce (enforce-dependency-convergence) @ pinot-transport ---
[INFO] --- maven-enforcer-plugin:3.0.0-M2:enforce (enforce-dependency-convergence) @ pinot-core ---
[INFO] --- maven-enforcer-plugin:3.0.0-M2:enforce (enforce-dependency-convergence) @ pinot-server ---
[INFO] --- maven-enforcer-plugin:3.0.0-M2:enforce (enforce-dependency-convergence) @ pinot-filesystem ---
[INFO] --- maven-enforcer-plugin:3.0.0-M2:enforce (enforce-dependency-convergence) @ pinot-controller ---
[INFO] --- maven-enforcer-plugin:3.0.0-M2:enforce (enforce-dependency-convergence) @ pinot-broker ---
[INFO] --- maven-enforcer-plugin:3.0.0-M2:enforce (enforce-dependency-convergence) @ pinot-api ---
[INFO] --- maven-enforcer-plugin:3.0.0-M2:enforce (enforce-dependency-convergence) @ pinot-hadoop ---
[INFO] --- maven-enforcer-plugin:3.0.0-M2:enforce (enforce-dependency-convergence) @ pinot-tools ---
[INFO] --- maven-enforcer-plugin:3.0.0-M2:enforce (enforce-dependency-convergence) @ pinot-minion ---
[INFO] --- maven-enforcer-plugin:3.0.0-M2:enforce (enforce-dependency-convergence) @ pinot-integration-tests ---
[INFO] --- maven-enforcer-plugin:3.0.0-M2:enforce (enforce-dependency-convergence) @ pinot-perf ---
[INFO] --- maven-enforcer-plugin:3.0.0-M2:enforce (enforce-dependency-convergence) @ pinot-hadoop-filesystem ---
[INFO] --- maven-enforcer-plugin:3.0.0-M2:enforce (enforce-dependency-convergence) @ pinot-distribution ---
```

File: pinot-integration-tests/pom.xml [hunk 1]
```diff
@@ -237,8 +241,21 @@
...
+        <exclusion>
+          <groupId>org.apache.zookeeper</groupId>
+          <artifactId>zookeeper</artifactId>
+        </exclusion>
       </exclusions>
     </dependency>
+    <dependency>
+      <groupId>com.sun.jersey</groupId>
+      <artifactId>jersey-core</artifactId>
+      <version>1.9</version>
```
Comment (jenniferdai) [inline pinot-integration-tests/pom.xml]: why did we add this dependency with a version? Should versions be enforced at the top level?

File: pom.xml [hunk 2]
```diff
@@ -696,9 +718,14 @@
       <dependency>
         <groupId>com.sun.jersey</groupId>
         <artifactId>jersey-client</artifactId>
-        <version>1.19.2</version>
+        <version>1.9</version>
```
Comment (jenniferdai) [inline pom.xml]: do all our apis/consoles still work with this change?

File: pom.xml [hunk 3]
```diff
@@ -729,8 +756,26 @@
...
+            <artifactId>javax.inject</artifactId>
+          </exclusion>
+          <exclusion>
+            <groupId>com.fasterxml.jackson.module</groupId>
+            <artifactId>ackson-module-jaxb-annotations</artifactId>
+          </exclusion>
         </exclusions>
       </dependency>
+      <dependency>
+        <groupId>com.fasterxml.jackson.jaxrs</groupId>
```
Comment (jenniferdai) [inline pom.xml]: What does this dependency do/why did we add these?

Comment (jackjlli) [reply pom.xml]: This dependency is brought from two sources, and we only need one:
```
Dependency convergence error for com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.5.4 paths to dependency are:
+-com.linkedin.pinot:pinot-server:0.016
  +-org.glassfish.jersey.media:jersey-media-json-jackson:2.24
    +-com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.5.4
and
+-com.linkedin.pinot:pinot-server:0.016
  +-io.swagger:swagger-jaxrs:1.5.10
    +-com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.4.5
```

File: pom.xml (hunk 2 above)
Comment (jackjlli) [reply pom.xml]: So `jersey-client` is marked as `test` and will only be used in the tests. I've already verified it.

File: pinot-integration-tests/pom.xml (hunk 1 above)
Comment (jackjlli) [reply pinot-integration-tests/pom.xml]: It seems `jersey-core` has been brought from hadoop-minicluster jar. We don't need to specify it here any more. I've removed it in the latest push. It's been tested.

**Comment by jackjlli** [comment]
@jenniferdai I've already tested it end to end.

File: pom.xml (hunk 3 above)
Comment (jenniferdai) [reply pom.xml]: does swagger still work when you make this change?

Comment (jackjlli) [reply pom.xml]: I've tested it. It works after the changes.

**Pull request merged**